sc-kpm/sc-agents-common/CMakeLists.txt

```diff
@@ -1,14 +1,22 @@
 set(SC_AGENTS_COMMON_SRC "${CMAKE_CURRENT_LIST_DIR}")
 
-file(GLOB_RECURSE SOURCES "*.cpp" "*.hpp")
+file(GLOB SOURCES CONFIGURE_DEPENDS
+    "*.cpp" "*.hpp"
+    "*/*.cpp" "*/*.hpp"
+)
 
-include_directories(${SC_AGENTS_COMMON_SRC} ${SC_MEMORY_SRC} ${SC_KPM_SRC})
+find_package(nlohmann_json 3.2.0 REQUIRED)
```

Comment:
It probably makes more sense to do this in the root CMakeLists.txt of the sc-machine  (unless sc-kpm is the only module that uses json, but I believe that at least sc-server and sc-ui also use nlohmann_json)